libtenzir/builtins/formats/xsv.cpp

```diff
@@ -633,6 +633,10 @@ auto parse_loop(generator<std::optional<std::string_view>> lines,
                 xsv_parser_options args) -> generator<table_slice> {
   // Parse header.
   auto it = lines.begin();
+  if (it == lines.end()) {
+    co_yield {};
+    co_return;
+  }
```

Comment:
I don't think that works, in particular because of our empty-thing-means-yield.

Reply:
I don't quite understand this comment. I think the `co_yield {}` is unnecessary, but other than that this looks good?

Reply:
These 4 lines were located at the top of the function when I commented this. Now that they are moved it should work as intended, I agree.